src/sql/base/browser/ui/table/table.ts

```diff
@@ -111,9 +115,38 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 
 		this.mapMouseEvent(this._grid.onContextMenu, this._onContextMenu);
 		this.mapMouseEvent(this._grid.onClick, this._onClick);
+		this.mapMouseEvent(this._grid.onHeaderClick, this._onHeaderClick);
 		this._grid.onColumnsResized.subscribe(() => this._onColumnResize.fire());
 	}
 
+	private invalidateRange(start: number, end: number): void {
+		let refreshedRows = range(start, end);
+		this._grid.invalidateRows(refreshedRows, true);
+		this._grid.render();
+	}
+
+	private renderGridRange(startIndex: number, count: number): void {
+		let editor = this._grid.getCellEditor();
+		let oldValue = editor ? (<Slick.Editors.Text<T>>editor).getValue() : undefined;
```

Comment:
you could assign `(<Slick.Editors.Text<T>>editor)` to a variable to avoid having to do the ugly cast twice in the function.

Reply:
Also just to keep in mind, typescript recommends using `as SomeInterface` instead of `<SomeInterface>` for type assertion. Although both syntaxes are correct, the latter could cause issues down the line if we need to use TSX.

Reply:
Addressed